src/utils/aquarius.ts

```diff
@@ -115,14 +116,22 @@ export async function transformDDOToAssetSelection(
   const didList: string[] = []
   const priceList: PriceList = await getAssetPrices(ddoList)
   const symbolList: any = {}
+  const didProviderEndpointMap: any = {}
   for (const ddo of ddoList) {
     didList.push(ddo.id)
     symbolList[ddo.id] = ddo.dataTokenInfo.symbol
+    const algoComputeService = ddo.findServiceByType('compute')
+    algoComputeService?.serviceEndpoint &&
+      (didProviderEndpointMap[ddo.id] = algoComputeService?.serviceEndpoint)
   }
   const ddoNames = await getAssetsNames(didList, metadataCacheUri, source.token)
   const algorithmList: AssetSelectionAsset[] = []
   didList?.forEach((did: string) => {
-    if (priceList[did]) {
+    if (
+      priceList[did] &&
+      (!didProviderEndpointMap[did] ||
+        didProviderEndpointMap[did] === datasetProviderEndpoint)
+    ) {
```

Comment:
I think we ca do this check before pushing the ddo id to `didList` so than we won't need the `didProviderEndpointMap`